subgraphs/balancer-v2/src/common/getters.ts

```diff
@@ -0,0 +1,185 @@
+import { Address, BigInt, dataSource, ethereum } from "@graphprotocol/graph-ts";
+import {
+  Token,
+  DexAmmProtocol,
+  LiquidityPool,
+  LiquidityPoolFee,
+  FinancialsDailySnapshot,
+  UsageMetricsDailySnapshot,
+  Swap,
+  PoolDailySnapshot,
+} from "../../generated/schema";
+
+import { fetchTokenSymbol, fetchTokenName, fetchTokenDecimals } from "./tokens";
+import {
+  BIGDECIMAL_ZERO,
+  Network,
+  VAULT_ADDRESS,
+  ProtocolType,
+  SECONDS_PER_DAY,
+  BIGINT_ZERO,
+  LiquidityPoolFeeType,
+} from "./constants";
+import { WeightedPool } from "../../generated/Vault/WeightedPool";
+import { ConvergentCurvePool } from "../../generated/Vault/ConvergentCurvePool";
+import {log} from "matchstick-as";
+
+export function getOrCreateDex(): DexAmmProtocol {
+  let protocol = DexAmmProtocol.load(VAULT_ADDRESS.toHexString());
+  let network = dataSource.network();
+  if (network === "polygon") {
+    network = Network.POLYGON;
+  } else {
+    network = Network.ETHEREUM;
+  }
+
+  if (protocol === null) {
+    protocol = new DexAmmProtocol(VAULT_ADDRESS.toHexString());
+    protocol.name = "Balancer V2";
+    protocol.schemaVersion = "0.0.2";
```

Comment:
update versions, include methodology version

Reply:
Will do when the latest schema update is merged :-D